pkg/clusteragent/admission/start.go

```diff
@@ -0,0 +1,78 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build kubeapiserver
+
+package admission
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/admission/controllers/secret"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/admission/controllers/webhook"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/common"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+// ControllerContext holds necessary context for the admission controllers
+type ControllerContext struct {
+	IsLeaderFunc func() bool
+	Informers    informers.SharedInformerFactory
+	Client       kubernetes.Interface
+	StopCh       chan struct{}
+}
+
+// StartControllers starts the secret and webhook controllers
+func StartControllers(ctx ControllerContext) error {
+	if !config.Datadog.GetBool("admission_controller.enabled") {
+		log.Info("Admission controller is disabled")
+		return nil
+	}
+
+	secretConfig := secret.Config{
```

Comment:
maybe we can create a function "NewConfig(...)", because all the `Config` fields have getters